### ISSUE TITLE
error matching routes with the same url and different methods

### ISSUE THREAD

**Issue by AlaaAttya** [skipperbent/simple-php-router#439]
for example 

```
SimpleRouter::get('/recipe', 'App\Application\Request\ListRecipesRequest@handle')->name("getRecipe");

SimpleRouter::post("/recipe", 'App\Application\Request\CreateRecipeRequest@handle')->name('createRecipe');
```

this shows this error

```
route "recipe" or method "get" not allowed
```

**Comment by Tahrz** [comment]
Hi! try to use `match()` method 

```
match(array $requestMethods, string $url, $callback, array $settings = null)
```

**Comment by skipperbent** [comment]
This issue has been resolved in version 4.3.0.0.
Feel free to open another issue if you are still experiencing issues.
